src/distilabel/llm/anthropic.py

```diff
@@ -0,0 +1,124 @@
+        try:
+            from anthropic import AsyncAnthropic
+        except ImportError as ie:
+            raise ImportError(
+                "Anthropic Python client is not installed. Please install it using"
+                " `pip install anthropic`."
+            ) from ie
+
+        self.api_key = self._handle_api_key_value(
+            self_value=self.api_key,
+            load_value=api_key,
+            env_var=self._env_var,  # type: ignore
+        )
+
+        self._aclient = AsyncAnthropic(
+            api_key=self.api_key.get_secret_value(),
+            base_url=self.base_url,
+            timeout=self.timeout,
+            http_client=self.http_client,
+            max_retries=self.max_retries,
+        )
+
+    @property
+    def model_name(self) -> str:
+        """Returns the model name used for the LLM."""
+        return self.model
+
+    async def agenerate(  # type: ignore
+        self,
+        input: "ChatType",
+        system: str = "",
+        num_generations: int = 1,
+        max_tokens: int = 128,
+        stop_sequences: List[str] = None,
+        temperature: float = 1.0,
+        top_p: float = 1.0,
+        top_k: int = 0,
+    ) -> "GenerateOutput":
+        """Generates a response asynchronously, using the [Anthropic Async API definition](https://github.com/anthropics/anthropic-sdk-python).
+
+        Args:
+            input: a single input in chat format to generate responses for.
+            system: the system prompt to use for the generation. No existing `system` role within the input conversation, only `user` and `assistant`. Defaults to `""`.
+            num_generations: the number of generations to create per input. Defaults to `1`.
+            max_tokens: the maximum number of new tokens that the model will generate. Defaults to `128`.
+            stop_sequences: custom text sequences that will cause the model to stop generating. Defaults to None.
+            temperature: the temperature to use for the generation. Set only if top_p is None. Defaults to `1.0`.
+            top_p: the top-p value to use for the generation. Defaults to `1.0`.
+            top_k: the top-k value to use for the generation. Defaults to `0`.
+
+        Returns:
+            A list of lists of strings containing the generated responses for each input.
+        """
+
+        completion = await self._aclient.messages.create(
+            model=self.model,
+            system=system,
+            messages=input,
+            max_tokens=max_tokens,
+            stream=False,
+            stop_sequences=stop_sequences,
+            temperature=temperature,
+            top_p=top_p,
+            top_k=top_k,
+        )
+        if (content := completion.content[0].text) is None:
+            self._logger.warning(
+                f"Received no response using Anthropic client (model: '{self.model}')."
+                f" Finish reason was: {completion.stop_reason}"
+            )
```

Comment:
As Anthropic AI client doesn't support an `n` or `num_generations` parameter, we should handle that on distilabel implementation overriding `generate` method like here: 

https://github.com/argilla-io/distilabel/blob/358333c6aa3b5ce8d9f55cf0e1f58c87cb59cdd7/src/distilabel/llm/mistral.py#L140-L164